ignite/metrics/rouge.py

```diff
@@ -0,0 +1,157 @@
+from collections import defaultdict
+from typing import Callable, DefaultDict, List, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics import Metric
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import reinit__is_reduced, sync_all_reduce
+
+
+class Rouge(Metric):
+    r"""Calculates the Rouge Score for two Sequence of Tokens.
+
+    Paper: Lin, Chin-Yew. 2004. ROUGE: a Package for Automatic Evaluation of Summaries.
+    In Proceedings of the Workshop on Text Summarization Branches Out (WAS 2004), Barcelona, Spain, July 25 - 26, 2004.
+
+
+    .. math::
+        F_\1 = 2 * \frac{\text{precision} * \text{recall}} {\text{precision} + \text{recall}}
```

Comment:
If f1 is used, remove useless alpha parameter.

But the paper relies on fbeta, so I think alpha should be renamed beta and implement fbeta formula rather that f1.

As discussed, we could think about variant later.